### PR TITLE
Change Sys_SnapVector to `rint` the components for x64 linux/mingw.

### PR DESCRIPTION
**Pull request by krsh732**
As far as I can tell, all the other platforms/engines round the vector
components to the nearest int (even). `round` doesn't do this,
regardless of the rounding mode that is set. `rint` on the other hand
does do this.

### CODE REVIEW

**Comment by ec-** [comment]
I see no differences between **round** a **rint** on x86/x86_64 and windows/linux hosts, is there any examples or explanations related to other engines/platforms?

**Comment by krsh732** [comment]
Here is an example: https://ideone.com/zHFrHZ

**Comment by ec-** [comment]
**round** works as expected there, **rint** do not, and we don't care about any other engine and their "improvements" because we must retain QVM compatibility, and QVM expects it to be rounded to nearest as it was always

**Comment by krsh732** [comment]
https://github.com/id-Software/Quake-III-Arena/blob/dbe4ddb10315479fc00086f08e25d968b4b43c49/code/unix/snapvector.nasm#L37

**Comment by ec-** [comment]
All x87 @ 0x037F, SSE and **round** functions produces exactly the same output for me on all platforms:
1.0 for 1.1
**2.0** for **1.5**
2.0 for 1.9

**Comment by krsh732** [comment]
TLDR: https://ideone.com/MyY2a8

In your case, all three functions will produce the same output because both the *nearest* and *nearest even* are 2.0...

Example program demonstrating discrepancy:
```c
#include <stdio.h>
#include <math.h>

#define QROUNDX87(src) \
	"flds " src "\n" \
	"fistpl " src "\n" \
	"fildl " src "\n" \
	"fstps " src "\n"

void q3_round( float *v )
{
	static const unsigned short cw037F = 0x037F;
	unsigned short cwCurr;

	__asm__ volatile
	(
		"fnstcw %1\n" \
		"fldcw %2\n" \
		QROUNDX87("0(%0)")
		"fldcw %1\n" \
		:
		: "r" (v), "m"(cwCurr), "m"(cw037F)
		: "memory", "st"
	);
}

int main(void) {
	// your code goes here
	float f = 4.5;
	q3_round(&f);
	printf("q3_round(4.5): %f\n", rint(4.5));
	printf("rint(4.5): %f\n", rint(4.5));
	printf("round(4.5): %f\n", round(4.5));
	return 0;
}
```
Results when running it:
```
q3_round(4.5): 4.000000
rint(4.5): 4.000000
round(4.5): 5.000000
```

**Comment by krsh732** [comment]
TLDR: https://ideone.com/MyY2a8

In your case, all three functions will produce the same output because both the *nearest* and *nearest even* are 2.0...

Example program demonstrating discrepancy:
```c
#include <stdio.h>
#include <math.h>

#define QROUNDX87(src) \
	"flds " src "\n" \
	"fistpl " src "\n" \
	"fildl " src "\n" \
	"fstps " src "\n"

void q3_round( float *v )
{
	static const unsigned short cw037F = 0x037F;
	unsigned short cwCurr;

	__asm__ volatile
	(
		"fnstcw %1\n" \
		"fldcw %2\n" \
		QROUNDX87("0(%0)")
		"fldcw %1\n" \
		:
		: "r" (v), "m"(cwCurr), "m"(cw037F)
		: "memory", "st"
	);
}

int main(void) {
	// your code goes here
	float f = 4.5;
	q3_round(&f);
	printf("q3_round(4.5): %f\n", f);
	printf("rint(4.5): %f\n", rint(4.5));
	printf("round(4.5): %f\n", round(4.5));
	return 0;
}

```
Results when running it:
```
q3_round(4.5): 4.000000
rint(4.5): 4.000000
round(4.5): 5.000000
```

**Comment by ec-** [comment]
Now I get the point, deleted your duplicated comment, thank you
